### PR TITLE
Improve dynamic color choices for bots

### PR DESCRIPTION
**Pull request by ruler501**
Calculates off of ELO, not count for openness and precomputes most of the values to avoid doing as many duplicate calculations.

### CODE REVIEW

File: src/utils/Draft.js [hunk 1]
```diff
@@ -281,6 +290,9 @@ async function finish() {
       const colors = botColors(null, null, draft.seats[i].pickorder, cards);
       draft.seats[i].name = `Bot ${i + 1}: ${colors.join(', ')}`;
       draft.seats[i].description = `This deck was drafted by a bot with color preference for ${colors.join('')}.`;
+    } else {
+      const colors = botColors(null, null, draft.seats[i].pickorder, cards);
+      draft.seats[i].name = `${draft.seats[i].name}: ${colors.join(', ')}`;
```
Comment (RKelln) [inline src/utils/Draft.js:295]: No description?

File: src/utils/Draft.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+import { COLOR_COMBINATIONS } from 'utils/Card';
 
 let draft = null;
 
+const toValue = (elo) => 10 ** (elo / 400);
+
+function addSeen(seen, cards) {
+  for (const card of cards) {
+    const colors = card.colors ?? card.details.colors ?? [];
+    if (colors.length > 0) {
```
Comment (RKelln) [inline src/utils/Draft.js:12]: No card rating needs to be added for colorless cards or do they have a color?  'C'?

**Comment by RKelln** [comment]
If you have time I'd like to see more comments added to describe some of the more complicated sections of the code. Pretty hard to read though quickly and understand what is going on and helpful for future maintenance / changes.

File: src/utils/Draft.js (hunk 1 above)
Comment (ruler501) [reply src/utils/Draft.js:306]: Description gets added by the user, we're just modifying it so the name includes our estimated colors of the deck like it does for the bot.

File: src/utils/Draft.js (hunk 2 above)
Comment (ruler501) [reply src/utils/Draft.js:12]: Colorless gets added to every combination so just reduces variance so they get skipped here.

**Pull request merged**